src/MatchedLearnerApi.Application/Mappers/MatchedLearnerDtoMapper.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MatchedLearnerApi.Application.Data.Models;
+using MatchedLearnerApi.Types;
+
+namespace MatchedLearnerApi.Application.Mappers
+{
+    public interface IMatchedLearnerDtoMapper
+    {
+        MatchedLearnerDto Map(List<DatalockEvent> datalockEvents);
+    }
+
+    public class MatchedLearnerDtoMapper : IMatchedLearnerDtoMapper
+    {
+        public MatchedLearnerDto Map(List<DatalockEvent> datalockEvents)
+        {
+            if (datalockEvents == null || !datalockEvents.Any())
+                return null;
+
+            var firstEvent = datalockEvents.First();
+            
+            return new MatchedLearnerDto
+            {
+                StartDate = firstEvent.LearningStartDate.GetValueOrDefault(),
+                EventTime = firstEvent.EventTime,
+                IlrSubmissionDate = firstEvent.IlrSubmissionDateTime,
+                IlrSubmissionWindowPeriod = firstEvent.CollectionPeriod,
+                AcademicYear = firstEvent.AcademicYear,
+                Ukprn = firstEvent.Ukprn,
+                Uln = firstEvent.LearnerUln,
+                Training = datalockEvents.Select(dataLockEvent => new TrainingDto
+                {
+                    Reference = dataLockEvent.LearningAimReference,
+                    ProgrammeType = dataLockEvent.LearningAimProgrammeType,
+                    StandardCode = dataLockEvent.LearningAimStandardCode,
+                    FrameworkCode = dataLockEvent.LearningAimFrameworkCode,
+                    PathwayCode = dataLockEvent.LearningAimPathwayCode,
+                    FundingLineType = null,
+                    StartDate = dataLockEvent.LearningStartDate.GetValueOrDefault(),
+                    //todo Status we think should be mapped on the price episode not on the Training
```

Comment:
If I remember right we decided not to include status, if this is the case this can be removed

Reply:
done